src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -99,12 +99,18 @@ def trace(self) -> JinjaTrace:
             alt_id, content_info, literal = value
             target_slice_idx = self.find_slice_index(alt_id)
             slice_length = content_info if literal else len(str(content_info))
-            if not is_set:
+            if not is_set_or_macro:
                 self.move_to_slice(target_slice_idx, slice_length)
             else:
-                # If we find output from a {% set %} directive, record a trace
-                # without reading or updating the program counter.
-                self.record_trace(slice_length, target_slice_idx)
+                # If we find output from a {% set %} directive or a macro,
+                # record a trace without reading or updating the program
+                # counter. Such slices are always treated as "templated"
+                # because they are inserted during expansion of templated
+                # code (i.e. {% set %} variable or macro defined within the
+                # file).
+                self.record_trace(
+                    slice_length, target_slice_idx, slice_type="templated"
+                )
```

Comment:
This fixes a different but related bug: Code inserted by using a `{% set %}` variable **or** a macro defined in the same file should be labelled as "templated". I discovered this when adding the L009 test case -- initially it was seeing literal newlines inserted by a macro as actual literals (bad!).